### PR TITLE
Add SearchManagerV2

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/Foundation/Connect/code/Managers/Search/ISearchManagerV2.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    using Sitecore.Data.Items;
+
+    /// <summary>
+    /// Performs search for products by specified parameters
+    /// </summary>
+    public interface ISearchManagerV2
+    {
+        /// <summary>
+        /// Get products from context catalog by keyword and search options
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Managers/Search/ISearchManagerV2.cs]: s

File: src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            Assert.Empty(results.SearchResultItems);
+        }
+
+        [Fact]
+        public void GetProducts_ShouldReturnNotNullSearchResults()
+        {
+            //arrange
+            var results = this.searchManager.GetProducts(
+                this.fixture.Create<string>(),
+                this.fixture.Create<SearchOptions>());
```
Comment (yuya-yuliya) [inline src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs]: This is Arrange and Act

File: src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+                        IQueryable<CommerceSellableItemSearchResultItem>>>());
+        }
+
+        [Fact]
+        public void GetProductItem_IfProductNotFound_ShouldReturnNull()
+        {
+            //arrange
+            var results = this.searchManager.GetProductItem(
+                this.fixture.Create<string>());
+
```
Comment (yuya-yuliya) [inline src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs]: Arrange and Act

File: src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+                    Arg.Any<Func<IQueryable<CommerceSellableItemSearchResultItem>,
+                        IQueryable<CommerceSellableItemSearchResultItem>>>());
+        }
+        
+        [Fact]
+        public void GetCategoryItem_IfCategoryNotFound_ShouldReturnNull()
+        {
+            //arrange
+            var results = this.searchManager.GetCategoryItem(
+                this.fixture.Create<string>());
```
Comment (yuya-yuliya) [inline src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs]: Arrange and Act

File: src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+                this.sitecoreContext,
+                this.commerceTypeLoader,
+                this.searchResponseProvider,
+                this.searchResultProvider);
+        }
+
+        #region GetProducts
+
+        [Fact]
+        public void GetProducts_IfArgumentNull_ShouldThrowArgumentNullException()
```
Comment (stanislav-kozorez) [inline src/Foundation/Connect/tests/Managers/Search/SearchManagerV2Tests.cs]: As an alternative you can use `[Theory]` with `[MemberData]` attribute

**Pull request merged**